pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -101,18 +114,24 @@ func (e *Executor) checkApproval(ctx context.Context) (string, bool) {
 		return "", false
 	}
 
-	metadata := map[string]string{
-		approvedByKey: approveCmd.Commander,
+	if err := approveCmd.Report(ctx, model.CommandStatus_COMMAND_SUCCEEDED, nil, nil); err != nil {
+		e.Logger.Error("failed to report handled command", zap.Error(err))
 	}
-	if err := e.MetadataStore.Stage(e.Stage.Id).PutMulti(ctx, metadata); err != nil {
-		e.LogPersister.Errorf("Unabled to save approver information to deployment, %v", err)
+
+	as := e.validateApproverNum(ctx, approveCmd.Commander, num)
```

Comment:
If you think about it, we don't need to save the last approver as metadata/
`validateApproverNum` is enough to return a bool indicating whether or not it got a sufficient number of approvals.

Reply:
So how about kind of like this
```go
	reached := e.validateApproverNum(ctx, approveCmd.Commander, num)
	if err := approveCmd.Report(ctx, model.CommandStatus_COMMAND_SUCCEEDED, nil, nil); err != nil {        
		e.Logger.Error("failed to report handled command", zap.Error(err))
	}
        return reached
```

Reply:
You can emit log inside `validateApproverNum` instead of L79-L80